synapse/handlers/sso.py

```diff
@@ -693,8 +697,49 @@ async def _register_mapped_user(
         await self._store.record_user_external_id(
             auth_provider_id, remote_user_id, registered_user_id
         )
+
+        # Set avatar, if available
+        if attributes.picture:
+            await self.set_avatar(registered_user_id, attributes.picture)
+
         return registered_user_id
 
+    async def set_avatar(self, user_id: str, picture_https_url: str) -> None:
+        try:
+            uid = UserID.from_string(user_id)
+
+            # download picture
+            http_response = requests.get(picture_https_url)
```

Comment:
We probably want to check the size of the request before doing the download, and limit it so we don't explode the RAM on the way. There is already a config for that, `max_avatar_size`.

Reply:
also, don't forget to check the blacklists, to make sure that someone can't fetch internal resources by supplying an avatar url like "http://localhost/sensitive_file". (The existing helpers may already do this, I haven't checked).

Reply:
@MatMaul I am now checking for max_avatar_size 👍 

@richvdh Do you mean the blacklists configured in config which are used when http client is created using [get_proxied_blacklisted_http_client()](https://github.com/matrix-org/synapse/blob/40fa8294e3096132819287dd0c6d6bd71a408902/synapse/server.py#L420)? Most likely not, since sensitive localhost urls should be blocked without any config. Are there other existing helpers that I can leverage?

Reply:
Ok cool, going to mark this as resolved then!